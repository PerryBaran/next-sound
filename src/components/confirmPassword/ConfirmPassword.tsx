"use client"

import { useRef } from "react"
import css from "./confirmPassword.module.css"

interface Props {
  callback: (string: string) => void
  setConfirm: (string: string) => void
}

export default function ConfirmPassword({ callback, setConfirm }: Props) {
  const password = useRef<HTMLInputElement>(null)

  const handleCancel = () => {
    setConfirm("")
  }

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault()
    if (password.current) {
      callback(password.current.value)
    }
  }

  return (
    <div className={css["container"]}>
      <form
        onSubmit={handleSubmit}
        aria-label="form"
        className={css["confirm-password"]}
      >
        <label htmlFor="confirm">
          <span>Confirm Password</span>
          <input type="password" id="confirm" ref={password} />
        </label>
        <div>
          <button type="submit" className={css["confirm"]}>
            Confirm
          </button>
          <button
            type="button"
            onClick={handleCancel}
            className={css["cancel"]}
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  )
}
