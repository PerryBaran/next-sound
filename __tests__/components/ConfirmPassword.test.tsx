import { fireEvent, render, screen } from "@testing-library/react"
import ConfirmPassword from "../../src/components/confirmPassword/ConfirmPassword"

describe("ConfirmPassword", () => {
  let props = {
    callback: jest.fn(),
    setConfirm: jest.fn()
  }

  beforeEach(() => {
    props = {
      callback: jest.fn(),
      setConfirm: jest.fn()
    }
  })

  test("renders correctly", () => {
    render(<ConfirmPassword {...props} />)

    expect(screen.getByRole("form")).toBeTruthy()
    expect(screen.getByLabelText(/confirm password/i)).toHaveAttribute("type", "password")
    expect(screen.getByText(/^confirm$/i)).toBeInstanceOf(HTMLButtonElement)
    expect(screen.getByText(/cancel/i)).toBeInstanceOf(HTMLButtonElement)
  })

  test("confirm button", () => {
    render(<ConfirmPassword {...props} />)

    const confirmButton = screen.getByText(/^confirm$/i)

    expect(props.callback).toBeCalledTimes(0)

    fireEvent.click(confirmButton)

    expect(props.callback).toBeCalledTimes(1)
    expect(props.callback).toBeCalledWith("")

    const password = "test"
    fireEvent.change(screen.getByLabelText(/confirm password/i), { target: { value: password }})
    fireEvent.click(confirmButton)

    expect(props.callback).toBeCalledTimes(2)
    expect(props.callback).toBeCalledWith(password)
  })

  test("cancel button", () => {
    render(<ConfirmPassword {...props} />)

    expect(props.setConfirm).toBeCalledTimes(0)

    fireEvent.click(screen.getByText(/cancel/i))

    expect(props.setConfirm).toBeCalledTimes(1)
    expect(props.setConfirm).toBeCalledWith("")
  })

  test("snapshot", () => {
    const { asFragment } = render(<ConfirmPassword {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })
})