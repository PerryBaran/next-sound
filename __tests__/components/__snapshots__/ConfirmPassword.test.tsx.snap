// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPassword snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      aria-label="form"
      class="confirm-password"
    >
      <label
        for="confirm"
      >
        <span>
          Confirm Password
        </span>
        <input
          id="confirm"
          type="password"
        />
      </label>
      <div>
        <button
          class="confirm"
          type="submit"
        >
          Confirm
        </button>
        <button
          class="cancel"
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
